src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/tests/test_acr_commands.py

```diff
@@ -3,48 +3,51 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
-# AZURE CLI ACR TEST DEFINITIONS
-#pylint: disable=line-too-long
+from azure.cli.tests import (ScenarioTest, JMESPathCheck, JMESPathCheckExists, NoneCheck,
+                             StorageAccountPreparer, ResourceGroupPreparer)
 
-from azure.cli.core.test_utils.vcr_test_base import (
-    ResourceGroupVCRTestBase,
-    JMESPathCheck,
-    NoneCheck
-)
 
-class ACRTest(ResourceGroupVCRTestBase):
+class AcrCommandsTests(ScenarioTest):
+    def test_check_name_availability(self):
+        # the chance of this randomly generated name has a duplication is rare
+        name = self.create_random_name('clireg', 50)
+        self.cmd('acr check-name -n {}'.format(name), checks=JMESPathCheck('nameAvailable', True))
 
-    def __init__(self, test_method):
-        super(ACRTest, self).__init__(__file__, test_method, resource_group='acr_resource_group')
-        self.registry_name_1 = 'acrtestregistry1'
-        self.registry_name_2 = 'acrtestregistry2'
-        self.storage_account_1 = 'acrteststorage1'
-        self.storage_account_2 = 'acrteststorage2'
-        self.location = 'southcentralus'
+    @ResourceGroupPreparer()
```

Comment:
I can add prefix support. I don't understand the second part of the comments. The name of the newly created resource group is sent in as the parameter to the test method, can you use it to refer to the resource?